File: app/src/main/java/com/example/hondanaoyuki/progressbarapp/MyProgressBar.java
package com.example.hondanaoyuki.progressbarapp;

import android.content.Context;
import android.widget.ProgressBar;

public class MyProgressBar extends ProgressBar {
    public MyProgressBar(Context context) {
        super(context);
    }

    @Override
    public void setProgress(int progress)
    {
        super.setProgress(progress);
        if(progress == this.getMax())
        {
            //Do stuff when progress is max
        }
    }
}

File: app/src/main/java/com/example/hondanaoyuki/progressbarapp/MyProgressBar.kt
package com.example.hondanaoyuki.progressbarapp

import android.content.Context
import android.widget.ProgressBar

class MyProgressBar(context: Context?) : ProgressBar(context) {

    override fun setProgress(progress: Int) {
        super.setProgress(progress)
        if (progress == this.max) {
            //Do stuff when progress is max
        }
    }
}